### ISSUE TITLE
关于token数量的疑问

### ISSUE THREAD

**Issue by DimplesL** [dvlab-research/MGM#33]
针对llama2-7b模型，max token length为2048，按照stage2 的训练参数，一旦设置IMAGE_GRID=2和IMAGE_GLOBAL为True，
                image_features = torch.cat([image_feat_global, image_features], dim=1)
                image_aux_features = torch.cat([image_aux_feat_global, image_aux_features], dim=1)
两行最终得到的图像特征token数就变成了2880，这里token数不是超了吗？如果理解有误敬请指正。

**Comment by DimplesL** [comment]
结合代码
        # Truncate sequences to max length as image embeddings can make the sequence longer
        tokenizer_model_max_length = getattr(self.config, 'tokenizer_model_max_length', None)
        if tokenizer_model_max_length is not None:
            new_input_embeds = [x[:tokenizer_model_max_length] for x in new_input_embeds]
            new_labels = [x[:tokenizer_model_max_length] for x in new_labels]


输入截断了token数量，至少是把扩展的视觉token截断了，不太理解这部分是怎么进行实验的

**Comment by DimplesL** [comment]
看了一下训练配置，是在stage2修改model_max_length为4096来实现的吗？这样修改对于llama没有其他影响吗？

**Comment by yanwei-li** [comment]
Hi, the max token size of [Vicuna-1.5](https://huggingface.co/lmsys/vicuna-7b-v1.5/blob/main/config.json) is 4K. So, it still within the limit.

**Comment by DimplesL** [comment]
> Hi, the max token size of [Vicuna-1.5](https://huggingface.co/lmsys/vicuna-7b-v1.5/blob/main/config.json) is 4K. So, it still within the limit.

I'm not very sure,  but as the training set in [train_vicuna_7b.sh,](https://github.com/lm-sys/FastChat/blob/main/scripts/train_vicuna_7b.sh) max length is 2048, which is the same set as llama2.
And the 4K, as shown in llama2, which means the context length, not only for input tokens' nums.

**Comment by yanwei-li** [comment]
Yes, it should include the output text. Here, we assume the total length should less than 4K.
